tests/testsuite/build_dir.rs

```diff
@@ -29,20 +29,23 @@ fn binary_with_debug() {
         )
         .build();
 
-    p.cargo("build").enable_mac_dsym().run();
+    p.cargo("-Zbuild-dir-new-layout build")
+        .masquerade_as_nightly_cargo(&["new build-dir layout"])
+        .enable_mac_dsym()
+        .run();
 
     assert_not_exists(&p.root().join("target"));
 
     p.root().join("build-dir").assert_build_dir_layout(str![[r#"
 [ROOT]/foo/build-dir/.rustc_info.json
-[ROOT]/foo/build-dir/CACHEDIR.TAG
-[ROOT]/foo/build-dir/debug/.cargo-lock
-[ROOT]/foo/build-dir/debug/.fingerprint/foo-[HASH]/bin-foo
-[ROOT]/foo/build-dir/debug/.fingerprint/foo-[HASH]/bin-foo.json
-[ROOT]/foo/build-dir/debug/.fingerprint/foo-[HASH]/dep-bin-foo
-[ROOT]/foo/build-dir/debug/.fingerprint/foo-[HASH]/invoked.timestamp
-[ROOT]/foo/build-dir/debug/deps/foo[..][EXE]
-[ROOT]/foo/build-dir/debug/deps/foo[..].d
+[ROOT]/foo/build-dir/[HOST_TARGET]/CACHEDIR.TAG
+[ROOT]/foo/build-dir/[HOST_TARGET]/debug/.cargo-lock
+[ROOT]/foo/build-dir/[HOST_TARGET]/debug/build/foo-[HASH]/deps/foo[..][EXE]
+[ROOT]/foo/build-dir/[HOST_TARGET]/debug/build/foo-[HASH]/deps/foo[..].d
+[ROOT]/foo/build-dir/[HOST_TARGET]/debug/build/foo-[HASH]/fingerprint/bin-foo
+[ROOT]/foo/build-dir/[HOST_TARGET]/debug/build/foo-[HASH]/fingerprint/bin-foo.json
+[ROOT]/foo/build-dir/[HOST_TARGET]/debug/build/foo-[HASH]/fingerprint/dep-bin-foo
+[ROOT]/foo/build-dir/[HOST_TARGET]/debug/build/foo-[HASH]/fingerprint/invoked.timestamp
```

Comment:
From @ranger-ross  at https://github.com/rust-lang/cargo/pull/15947#discussion_r2402129580

> I spent some time getting up to speed on Windows path issues. (goodness, its a deep rabbit hole)
> 
> ...
>
> Given the long paths issue on windows, perhaps it would be better to shorten names to help mitigate this? `build-script-execution` while descriptive is pretty long.